workshops/read_the_internet/README.md

```diff
@@ -0,0 +1,175 @@
+
+```python
+from bs4 import BeautifulSoup as bs
+import requests
+```
+
+Now we can start webscraping! We will begin by requesting our webpage and extracting the HTML from the request object.
+
+```python
+url = 'https://ncov2019.live/data'
+# request the website
+r = requests.get(url)
+# get the HTML from the request
+page = r.text
+```
+
+Next we will create the BeautifulSoup object and use `lxml` to parse the HTML.
+
+```python
+soup = bs(page, 'lxml')
+```
+
+Now that we have our BeautifulSoup object we can go all of the HTML! 
+
+### Working with the parsed HTML
+
+![HTML gif](https://cloud-5bmb4t5tp.vercel.app/2html.gif)
+
+In the next step, we'll be working with data from an HTML table. So, before, we continue, a quick mini-lesson on HTML tables:
+
+Here's an example of an HTML table:
+
+```html
+<table>
+    <tr>
+        <td>Country</td>
+        <td>Cases</td>
+        <td>Column</td>
+        <td>Column 2</td>
+    </tr>
+    <tr>
+        <td>United States</td>
+        <td>1,000,000</td>
+        <td>Column data</td>
+        <td>Column 2 data</td>
+    </tr>
+    <tr>
+        <td>Canada</td>
+        <td>100,000</td>
+        <td>Column data</td>
+        <td>Column 2 data</td>
+    </tr>
+</table>
+```
+
+A HTML table can be seen as a 2 dimensional array or list. The table contains rows (`tr` tags). Each row contains columns (`td` tags). Within the `td` tags is the data or text we can webscrape.
+
+### Python time!
+
+We're going to use BeautifulSoup to find an HTML table element. We want to find a specific table by ID `sortable_table_world` as seen in the picture. The code below finds the first table that matches that ID by using the `find` function and specifying the id as an attribute. The `attrs` is an optional argument and is a dictionary that can be set to any id, class, or HTML identifier.
+
+![nCOVID table open with inspect element](https://cloud-p5pdve0kf.vercel.app/1inspect_element.png)
+
+```python
+# get world data table
+table = soup.find('table', attrs={'id':'sortable_table_world'})
+```
+After this code runs we now have a new beautiful soup object called `table` that contains the table and child elements within that table.
+
+With `table` we will use the `findAll` function to get all the `tr` tags from the table. `tableRows` is a Python list with a beautiful soup object for each `tr` tag found in the table.
+```python
+# get table rows
+tableRows = table.findAll('tr')
+```
+Now that we have all the rows in a Python list, we can iterate through the list and try to find the United States or you can input any Country you want.
+
+Before getting to the looping part though we will need to take a look at some of our outputs. This may happen to you when webscraping, the text has weird formatting. In these next steps I'm going to show the text from the website with the weird formatting and how we can use Python to remove that.
```

Comment:
this paragraph is also kinda awkward wording and i don't feel like i'm fully understanding/picturing what i should be understanding/picturing after i read it. should be made clearer